### PR TITLE
Fix tel

### PR DESCRIPTION
**Pull request by mutsinziisaac**
**PR Description**
This PR refactors the user model to save the country code alongside the phone number and updates the relevant files accordingly.

**Description of tasks that were expected to be completed**

- Refactor the user model to combine country code and phone number.
- Update files affected by this change (especially those involving user communication).

Previously, the country code and phone number were stored separately in MongoDB, which created a poor user experience for managers and admins when communicating or retrieving a trainee's phone number.

**How has this been tested?**

1. Clone the repo.
2. Checkout the branch fix-tel.
3. Set up the environment variables as indicated in .env.example.
4. Run npm install followed by npm run dev to ensure the app runs successfully.

 **Deployed Link:**

N/A

**Track PR (Issue number & link):**
N/A

Screenshots (If appropriate):
N/A

### CODE REVIEW

**Review by Mugisha146 — APPROVED** [review]
Great job

**Review by robsdagreat — APPROVED** [review]

**Review by Philimuhire — APPROVED** [review]

**Review by mugishaj092 — APPROVED** [review]

**Review by SaddockAime — APPROVED** [review]

**Comment by ncutixavier** [comment]
@mutsinziisaac  You have many unnecessary changes

File: src/components/form/RegisterForm.tsx [hunk 1]
```diff
@@ -341,7 +347,11 @@ function SignupForm() {
                   </Button>
                 </>
               ) : (
-                <Button type="submit" label="Signup" className="my-1  mb-4 sm:w-full w-5/6 " />
+                <Button
+                  type="submit"
+                  label="Signup"
+                  className="my-1  mb-4 sm:w-full w-5/6 "
+                />
```
Comment (ncutixavier) [inline src/components/form/RegisterForm.tsx]: Unnecessary formatting

**Review by ncutixavier — COMMENTED** [review]
Please disable your formatter to avoid unnecessary changes

**Comment by Salim-54** [comment]
I can't merge this 
image

You should make sure that if I select a country I am going to get a suggestion of the relevant country code
and drop down of the country code suggests the country as the country dropdown which is very incorrect.
